connection.go

```diff
@@ -123,18 +156,69 @@ func (h *handler) handleConnection() {
 	go func() {
 		// TODO: propagate error
 		n, _ := io.Copy(h.conn, h.ptmx)
-		log.Printf("[handler] copied %d bytes to conn\n", n)
+		log.Printf("[connectionHandler] copied %d bytes to conn\n", n)
 	}()
 	go func() {
 		// TODO: propagate error
-		n, _ := io.Copy(h.ptmx, h.conn)
-		log.Printf("[handler] copied %d bytes to to ptmx\n", n)
+		err := <-h.routePackets()
```

Comment:
I don't really understand what's happening here, where is the `routePackets()` coming from?

Reply:
This boi here https://github.com/smowafy/insecure-shell/pull/2/files/83c4e311922c5ad3ba9f41adff234d668fbc9529#diff-25a7cf08cc8fc8eb57475835cc380524de2fd1215cda9c6915132ae86dbd80a7R175, it runs a goroutine with an infinite loop reading packets and returns the first error it gets on the channel